subtests/docker_cli/build/build.py

```diff
@@ -114,213 +124,191 @@ def dockerfile_repo_replace(self, path, repo):
         self.sub_stuff['di'] = dimg = DockerImages(self)
         self.sub_stuff['existing_images'] = dimg.list_imgs_ids()
-        img_name = dimg.get_unique_name()
-        # Build definition:
-        # build['image_name'] - name
-        # build['dockerfile_dir_path'] - path to docker file directory
-        # build['result'] - results of docker build ...
-        # build['intermediary_containers'] - Please set to true when --rm=False
-        # build['use_config_repo'] - Whether use repo 'docker_repo_name'
-        # build['dockerfile_path'] - path to Dockerfile
-        # build['base_repo_fqin'] - repo name in 'docker_rpo_name'
-        build_def = {}
-        self.sub_stuff['builds'] = [build_def]
-        build_def['image_name'] = img_name
-        if 'dockerfile_use_config_repo' not in self.config:
-            use_config_repo = 'net_dockerfile'
-        else:
-            use_config_repo = self.config.get('dockerfile_use_config_repo')
-        build_def['use_config_repo'] = use_config_repo
-        dir_path = self.config.get('dockerfile_dir_path')
-        if not dir_path:
-            raise DockerTestNAError(
-                "config['dockerfile_dir_path'] not provided")
-        build_def['dockerfile_dir_path'] = self.dockerfile_dir_path(dir_path)
-        if build_def['use_config_repo'] != 'net_dockerfile':
-            file_path = build_def['dockerfile_dir_path'] + '/' + 'Dockerfile'
-            if not os.path.isfile(file_path):
-                raise DockerTestNAError("Dockerfile does not exist")
-            build_def['dockerfile_path'] = file_path
-        else:
-            build_def['dockerfile_path'] = ''
-        im_cnt = self.config.get('docker_build_intermediary_containers')
-        build_def['intermediary_containers'] = im_cnt
-        build_def['build_fail_msg'] = self.config.get('docker_build_fail_msg')
-        build_def['stdout'] = self.config.get('docker_build_stdout')
-        build_def['no_stdout'] = self.config.get('docker_build_no_stdout')
-        full_repo = DockerImage.full_name_from_defaults(self.config)
-        build_def['full_repo'] = full_repo
-        if build_def['dockerfile_path']:
-            self._initialize_repo(build_def['use_config_repo'],
-                                  build_def['dockerfile_path'],
-                                  build_def['full_repo'])
+
+    def make_builds(self, source):
+        dimg = self.sub_stuff['di']
+        image_name = dimg.get_unique_name()
+        intermediate_containers = source['intermediate_containers']
+        additional_containers = source['additional_containers']
+        positive_build_regex = source['positive_build_regex'].strip()
+        negative_build_regex = source['negative_build_regex'].strip()
+        base_repo_fqin = DockerImage.full_name_from_defaults(source)
+        use_config_repo = source['use_config_repo']
+        dockerfile_dir_path = source['dockerfile_dir_path'].strip()
```

Comment:
Looks like dockerfile_dir_path lack of the system path which get from  self.parent_subtest.srcdir from func def dockerfile_dir_path, it's /full(for example) in my env now....